File: src/main/java/wraith/fabricaeexnihilo/modules/base/EnchantableBlockItem.java
package wraith.fabricaeexnihilo.modules.base;

import net.minecraft.block.Block;
import net.minecraft.item.BlockItem;
import net.minecraft.item.ItemStack;

public class EnchantableBlockItem extends BlockItem {
    
    private final int enchantability;
    
    public EnchantableBlockItem(Block block, Settings settings, int enchantability) {
        super(block, settings);
        this.enchantability = enchantability;
    }
    
    @Override
    public int getEnchantability() {
        return enchantability;
    }
    
    @Override
    public boolean isEnchantable(ItemStack stack) {
        return true;
    }
    
}

File: src/main/kotlin/exnihilofabrico/modules/base/EnchantableBlockItem.kt
package exnihilofabrico.modules.base

import net.minecraft.block.Block
import net.minecraft.item.BlockItem
import net.minecraft.item.ItemStack

class EnchantableBlockItem(block: Block, val _enchantability: Int, settings: Settings): BlockItem(block, settings) {

    override fun isEnchantable(stack: ItemStack?) = true
    override fun getEnchantability(): Int {
        return _enchantability
    }
}